docs/src/main/sphinx/security/user-mapping.rst

```diff
@@ -2,67 +2,123 @@
 
+.. _pattern-file:
+
 File mapping rules
 ------------------
 
-The file mapping rules allow for more complex mappings from the authentication user.  These rules are
-loaded from a JSON file defined in a configuration property.  The mapping is based on the first matching
-rule, processed from top to bottom. If no rules match, authentication is denied.  Each rule is composed of the
+Use the **File mapping rules** method if your authentication provider’s way to
+express usernames cannot be reduced to a single rule, or if you want to exclude
+a set of users from accessing the cluster.
+
+The rules are loaded from a JSON file identified in a configuration property.
+The mapping is based on the first matching rule, processed from top to bottom.
+If no rules match, authentication is denied.  Each rule is composed of the
 following fields:
 
-* ``pattern`` (required): regex to match against authentication user.
-* ``user`` (optional): replacement string to substitute against pattern.
+* ``pattern`` (required): regex to match against the authentication method's
+  username.
+* ``user`` (optional): replacement string to substitute against *pattern*.
   The default value is ``$1``.
-* ``allow`` (optional): boolean indicating if the authentication should be allowed.
+* ``allow`` (optional): Boolean indicating whether authentication is to be
+  allowed for the current match.
 * ``case`` (optional): one of:
 
-  * ``keep`` - keep matched user name as is (default behavior)
-  * ``lower`` - lowercase matched user name, e.g., ``Admin`` or ``ADMIN`` will become ``admin``
-  * ``upper`` - uppercase matched user name, e.g., ``admin`` or ``Admin`` will become ``ADMIN``
+  * ``keep`` - keep the matched username as is (default behavior)
+  * ``lower`` - lowercase the matched username; thus both ``Admin`` and ``ADMIN`` become ``admin``
+  * ``upper`` - uppercase the matched username; thus both ``admin`` and ``Admin`` become ``ADMIN``
 
-The following example maps all users like ``alice@example.com`` to just ``alice``, except for the ``test``
-user which is denied authentication, and it maps users like ``bob@uk.example.com`` to ``bob_uk``:
+The following example maps all usernames in the form ``alice@example.com`` to
+just ``alice``, except for the ``test`` user, which is denied authentication. It
+also maps users in the form ``bob@uk.example.com`` to ``bob_uk``:
 
 .. literalinclude:: user-mapping.json
     :language: json
 
-Each authentication system has a separate property for the user mapping file to allow different mapping
-when multiple authentication systems are enabled:
+For this example file placed in the installation's ``etc`` directory, configure
+it for an :doc:`LDAP </security/ldap>` authentication provider (which uses the
+PASSWORD authentication type), as follows:
+
+.. code-block:: text
+
+    http-server.authentication.password.user-mapping.file=etc/user-mapping.json
+
+You can place your user mapping JSON file in any file system location on the
```

Comment:
I'd remove "file system" - that's a given

Reply:
I meant to specify "local file system," which is not as given as you might suspect. It's very common for Unix-Linux servers to mount a dozen or more shares from other servers all over the network. It could cause all sorts of hard-to-diagnose errors if an admin put the user mapping file on a remote share on an unstable mounted disk that appears and disappears. Presto user auth might succeed or fail as the user mapping file appeared and disappeared. 

Now, the above presumes that Presto rereads the file for every user auth instance. But now that I think of it, we must be smarter than that. Surely we read the user mapping file once at server init and cache it for reference thereafter. But even if so, the server might either (a) fail to start if the file is momentarily not found on an unstable remote disk, or (b) start anyway, but fail to authenticate any users. Both of these cases would result in failures that would be very hard to diagnose. Thus: put the user mapping file on the coordinator's _local file system_.